### ISSUE TITLE
[eslint-plugin-lit-a11y]: implement no-static-element-interactions

### ISSUE THREAD

**Issue by bennypowers** [open-wc/open-wc#1930]
See: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-static-element-interactions.md

# no-static-element-interactions

Static HTML elements do not have semantic meaning. This is clear in the case of `<div>` and `<span>`. It is less so clear in the case of elements that _seem_ semantic, but that do not have a semantic mapping in the accessibility layer. For example `<a>`, `<big>`, `<blockquote>`, `<footer>`, `<picture>`, `<strike>` and `<time>` -- to name a few -- have no semantic layer mapping. They are as void of meaning as `<div>`.

The [WAI-ARIA `role` attribute](https://www.w3.org/TR/wai-aria-1.1/#usage_intro) confers a semantic mapping to an element. The semantic value can then be expressed to a user via assistive technology.

In order to add interactivity such as a mouse or key event listener to a static element, that element must be given a role value as well.

## Rule details

You may configure which handler props should be taken into account when applying this rule. The recommended configuration includes the following 6 handlers.

```javascript
'lit-a11y/no-static-element-interactions': ['error', {
    customElements: [
      'custom-input'
    ],
    events: [
      'click',
      'mousedown',
      'mouseup',
      'keypress',
      'keydown',
      'keyup',
    ],
],
```

Adjust the list of event names in the events array to add custom events and increase or decrease the coverage surface of this rule in your codebase.

### Succeed

```js
html`
  <button @click=${() => {}} className="foo"></button>
  <div className="foo" @click=${() => {}} role="button"></div>
  <input type="text" @click=${() => {}} />
`;
```

### Fail

```js
html`
  <div @click=${() => {}}></div>
`;
```

## Accessibility guidelines
- [WCAG 4.1.2](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value)

### Resources
- [WAI-ARIA `role` attribute](https://www.w3.org/TR/wai-aria-1.1/#usage_intro)
- [WAI-ARIA Authoring Practices Guide - Design Patterns and Widgets](https://www.w3.org/TR/wai-aria-practices-1.1/#aria_ex)
- [Fundamental Keyboard Navigation Conventions](https://www.w3.org/TR/wai-aria-practices-1.1/#kbd_generalnav)
- [Mozilla Developer Network - ARIA Techniques](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_button_role#Keyboard_and_focus)
